Client/game_sa/CBuildingSA.cpp

```diff
@@ -11,8 +11,56 @@
 
 #include "StdInc.h"
 #include "CBuildingSA.h"
+#include <game/CWorld.h>
+#include "CGameSA.h"
+
+extern CGameSA* pGame;
 
 CBuildingSA::CBuildingSA(CBuildingSAInterface* pInterface)
 {
     SetInterface(pInterface);
 }
+
+void CBuildingSA::SetLod(CBuilding* pLod)
+{
+    if (pLod)
+    {
+        if (m_pInterface->m_pLod)
+        {
+            SetLod(nullptr);
+        }
+
+        CBuildingSAInterface* pLodInterface = (dynamic_cast<CBuildingSA*>(pLod))->GetBuildingInterface();
+
+        // We should recreate buildings...
+        pGame->GetWorld()->Remove(pLodInterface, CBuilding_SetLod);
+        pGame->GetWorld()->Remove(m_pInterface, CBuilding_SetLod);
+
+        m_pInterface->m_pLod = pLodInterface;
+        pLodInterface->bUsesCollision = 0;
+        pLodInterface->numLodChildren = 1;
+
+        if (pGame->GetModelInfo(pLodInterface->m_nModelIndex)->GetLODDistance() > 300)
```

Comment:
Are you sure that GetModelInfo is always return a non-nullptr pointer?

Reply:
Yeah. All buildings and objects should have valid model ID. Otherwise, I deserve random crashes.